### PR TITLE
Add yaml linter in GitHub actions

### PR DESCRIPTION
**Pull request by gciavarrini**
This PR adds `yamllinter` to GitHub actions and fixes formatting problems for non-generated yaml files in project.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -57,6 +57,15 @@ jobs:
...
 
+  YamlLint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: ⤵️ Check out code from GitHub
+        uses: actions/checkout@v2
+      - name: 🚀 Run yamllint
+        run: |
+          yamllint -f github -c .yamllint .
+
```
Comment (gciavarrini) [inline .github/workflows/go.yml:69]: @eloycoto I'd like to have your feedback on this. :)

File: docs/design/http_api_swagger.md [hunk 2]
```diff
@@ -583,7 +583,9 @@ Status: Internal Server Error
 
 | Name | Type | Go type | Required | Default | Description | Example |
 |------|------|---------|:--------:| ------- |-------------|---------|
-| authFile | string| `string` |  | | Image registries authfile created by executing `podman login` or `docker login` (i.e. ~/.docker/config.json). https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#log-in-to-docker-hub describes how the file can be created and how it is structured. |  |
+| authFile | [interface{}](#interface)| `interface{}` |  | | Image registries authfile created by executing `podman login` or `docker login` (i.e. ~/.docker/config.json).
```
Comment (jakub-dzon) [inline docs/design/http_api_swagger.md:586]: The rendered MD formatting seems broken here - probably due to swagger.yaml formatting changes.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]

File: docs/design/http_api_swagger.md (hunk 2 above)
Comment (gciavarrini) [reply docs/design/http_api_swagger.md]: uh right! I accidentally deleted `type: string` from `swagger.yaml`

Comment (jakub-dzon) [reply docs/design/http_api_swagger.md]: It didn't help; it's still ill-formatted.

**Review by jakub-dzon — APPROVED** [review]

**Review by masayag — APPROVED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (eloycoto) [reply .github/workflows/go.yml:69]: all good!

**Pull request merged**